### DISCUSSION TITLE
Pocketbase S3 Bucket Confusion

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#3835] [Q&A] [answered]
Hey,

I added to my pocketbase the S3 Bucket to save images and videos for the collection "posts". 

Everything works fine and the images are saved correctly. However, the images are also stored locally on my machine which I tried to prevent in the first place.

Am I doing something wrong or did I misunderstood something

![Screenshot from 2023-11-28 17-29-05](https://github.com/pocketbase/pocketbase/assets/22890183/dc06fa32-7f29-48a8-aa5a-ec32991b766f)

**Comment by ganigeorgiev** [comment]
There is really not enough details to help you debug it.

>  However, the images are also stored locally on my machine

How did you confirm this? Are you sure that it is not just some client-side cache?

How did you setup the S3 storage? Note that there are 2 S3 configurations - 1 for the file uploads and a separate one for the backups.

**Reply by MuslemRahimi** [reply]
Ahhh sorry,

So the preview is downloading from the aws url. I checked it by closing the browser and cutting of the internet connection. 
Now the preview is not showing the image anymore.

Is that so far correct how you designed the process?

**Reply by ganigeorgiev** [reply, marked as answer]
I'm not sure that I understand the question.

When the S3 files storage is enabled, all record file operations will be performed via the S3 APIs using the configured credentials.

For example, when you hit `https://example.com/api/files/:collection/:record/:filename` the PocketBase server will request the specific file from S3 and then return the result blob back to the client.
